### PR TITLE
Worker thread affinity control

### PR DESCRIPTION
**Pull request by dostos**

### CODE REVIEW

File: ruy/ctx.h [hunk 1]
```diff
@@ -48,7 +49,8 @@ class Ctx /* not final, subclassed by CtxImpl */ {
   ThreadPool* mutable_thread_pool();
   int max_num_threads() const;
   void set_max_num_threads(int value);
-  CpuInfo* mutable_cpuinfo();
+  void set_cpu_mask(cpu_set_t cpu_mask);
+  CpuInfo *mutable_cpuinfo();
```
Comment (kdh0102) [inline ruy/ctx.h]:
```suggestion
  CpuInfo* mutable_cpuinfo();
```

File: ruy/thread_pool.h [hunk 2]
```diff
@@ -20,10 +20,12 @@ limitations under the License.
 #define RUY_RUY_THREAD_POOL_H_
 
 #include <vector>
+#include <sched.h> 
 
 #include "ruy/blocking_counter.h"
 #include "ruy/time.h"
 
+
```
Comment (kdh0102) [inline ruy/thread_pool.h]:
```suggestion
```

File: ruy/cpu.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// specific language governing permissions and limitations under the License.
+
+#ifndef RUY_RUY_CPU_H_
+#define RUY_RUY_CPU_H_
+
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <vector>
+#include <limits.h>
```
Comment (kdh0102) [inline ruy/cpu.h]:
```suggestion
#include <limits.h>
#include <vector>
```

**Review by kdh0102 — APPROVED** [review]
LGTM! left only a few style checks.
Also, I rebased the branch to `tf_v2.3.0` (the latest commit is what TF v.2.3.0 is using).

**Comment by dostos** [comment]
Thanks for the update!

**Pull request merged**